src/librustc_typeck/check/mod.rs

```diff
@@ -5387,7 +5387,9 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             _ => return None,
         };
         let last_expr_ty = self.node_ty(last_expr.hir_id);
-        if self.can_sub(self.param_env, last_expr_ty, expected_ty).is_err() {
+        if last_expr_ty.is_error()
```

Comment:
Instead of adding a method change this to `if matches!(last_expr_ty.kind, ty::Error)`

Reply:
Changed it